src/foam/lib/formatter/JSONFObjectFormatter.java

```diff
@@ -303,7 +302,16 @@ protected Boolean maybeOutputProperty(FObject fo, PropertyInfo prop, boolean inc
   }
 
   public void outputDelta(FObject oldFObject, FObject newFObject) {
-    ClassInfo info           = oldFObject.getClassInfo();
+    outputDelta(oldFObject, newFObject, null);
+  }
+
+  public void outputDelta(FObject oldFObject, FObject newFObject, ClassInfo defaultClass) {
+    ClassInfo info = newFObject.getClassInfo();
+
+    boolean outputClass = false;
+
+    if ( outputClassNames_ && ( info != defaultClass || outputDefaultClassNames_ ) ) outputClass = true;
```

Comment:
You could combine this line and the one above:
boolean outputClass = outputClassNames_ && ( info != defaultClass || outputDefaultClassNames_ );